### PR TITLE
Add occupancy sensor to the all-clusters-app

### PR DESCRIPTION
**Pull request by nivi-apple**
#### Problem
What is being fixed?  Examples:
* Adds occupancy sensor default value

#### Change overview
Add support for occupancy sensor

#### Testing
How was this tested? (at least one bullet point required)
* Tested using m5stack

### CODE REVIEW

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 1]
```diff
@@ -507,6 +507,14 @@ void SetupPretendDevices()
...
     AddCluster("Door");
     AddAttribute("State", "Closed");
+
+    AddDevice("Occupancy Sensor");
+    AddEndpoint("External");
+    AddCluster("Occupancy Sensor");
+    AddAttribute("Occupancy", "1");
+    uint8_t attributeValue = 1;
+    emberAfWriteServerAttribute(1, ZCL_OCCUPANCY_SENSING_CLUSTER_ID, ZCL_OCCUPANCY_ATTRIBUTE_ID, (uint8_t *) &attributeValue,
+                                ZCL_BITMAP8_ATTRIBUTE_TYPE);
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/main.cpp:517]: Please use Accessors.h functions for writing attributes.

Comment (dhrishi) [reply examples/all-clusters-app/esp32/main/main.cpp:517]: Why is the value written? For demonstration?

Comment (dhrishi) [reply examples/all-clusters-app/esp32/main/main.cpp:517]: Instead should we read the default value and then call `AddAttribute("Occupancy", value);`?

Comment (nivi-apple) [reply examples/all-clusters-app/esp32/main/main.cpp:517]: @dhrishi Yes for demo. We could do that.

Comment (nivi-apple) [reply examples/all-clusters-app/esp32/main/main.cpp]: @dhrishi getting the default value didn't work. so i just set i to occupied for now

File: examples/all-clusters-app/esp32/main/main.cpp [hunk 2]
```diff
@@ -507,6 +507,12 @@ void SetupPretendDevices()
     AddEndpoint("Door 2");
     AddCluster("Door");
     AddAttribute("State", "Closed");
+
+    AddDevice("Occupancy Sensor");
+    AddEndpoint("External");
+    AddCluster("Occupancy Sensor");
+    AddAttribute("Occupancy", "1");
+    chip::app::Clusters::OccupancySensing::Attributes::Occupancy::Set(1, 1);
```
Comment (bzbarsky-apple) [inline examples/all-clusters-app/esp32/main/main.cpp:515]:
```suggestion
    app::Clusters::OccupancySensing::Attributes::Occupancy::Set(1, 1);
```
File has `using namespace ::chip;`.

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/all-clusters-app/esp32/main/main.cpp (hunk 1 above)
Comment (dhrishi) [reply examples/all-clusters-app/esp32/main/main.cpp]: Right. The default value would  be possible/applicable for clusters like on-off, level control

**Review by dhrishi — APPROVED** [review]

**Comment by andy31415** [comment]
@nivi-apple - could you edit the summary to not say 'examples'? also `Fixes rdar://74031737` ... is this maybe an internal tracking reference? Could we replace/drop the line since others OSS developers do not know what the RDAR item contains.

**Review by andy31415 — APPROVED** [review]

**Comment by mspang** [comment]
I don't really understand this. This adds to the menu, but do the menu items do anything?

**Review by woody-apple — APPROVED** [review]

**Comment by andy31415** [comment]
@nivi-apple - I am merging this because it has checkmarks and change is trivial.
However I do concur with the comment from @mspang - this seems to just add a menu, I am unclear how this is functional.

**Pull request merged**